.buildkite/scripts/grapl_testing_stack/destroy_stateful_resources.sh

```diff
@@ -48,14 +48,26 @@ urns=$(pulumi stack output stateful-resource-urns --stack="${stack}")
 mapfile -t target_args < <(echo "${urns}" | jq -r '. | map("--target=" + .) | .[]')
 
 echo -e "--- :pulumi: Destroying stateful resources for ${stack}"
-pulumi destroy \
-    --cwd="${project_dir}" \
-    --stack="${stack}" \
-    --show-replacement-steps \
-    --non-interactive \
-    --diff \
-    --yes \
-    --refresh \
-    --message="Destroying stateful resources" \
-    --target-dependents \
-    "${target_args[@]}"
+
+# Attempt to destroy each target, one at a time.
+# This can be de-for-loop-ified after
+# https://github.com/pulumi/pulumi/issues/3304
+for target_arg in "${target_args[@]}"; do
+    (
+        # Allow errors; this is best-effort and should allow for failed
+        # destroys of targets that no longer exist.
+        set +e
+
+        pulumi destroy \
+            --cwd="${project_dir}" \
+            --stack="${stack}" \
+            --show-replacement-steps \
+            --non-interactive \
+            --diff \
+            --yes \
+            --refresh \
+            --message="Destroying stateful resources" \
+            --target-dependents \
+            "${target_arg}"
+    )
+done
```

Comment:
will the subshell work the way I want it to here?
I want to allow any failures inside the () to be okay; but I'm worried it'll be like "oh the subshell returned nonzero, so, that's a Command That Failed"

Reply:
tested it and sure enough this never gets to hey :) 
```
set -euo pipefail
(
  set +e
  exit 1
)
echo "hey"
```